app/db/models.py

```diff
@@ -1,22 +1,68 @@
 """Database models for the application."""
 
+from typing import Any
 from uuid import uuid4
 
-from sqlalchemy import String
+from loguru import logger
+from sqlalchemy import String, event, text
 from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.engine import Connection
+from sqlalchemy.exc import SQLAlchemyError
 from sqlalchemy.orm import Mapped, mapped_column
 
 from app.db.base import Base
 from app.db.model_mixins import TimestampMixin
 
+NOTIFICATION_STARTING_PARTITION_YEAR = 2024
+NOTIFICATION_PARTITION_DURATION_YEARS = 10
+
 
 class Notification(TimestampMixin, Base):
     """Database table for notifications."""
 
     __tablename__ = 'notifications'
-
-    id: Mapped[UUID[str]] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid4)
-    personalization: Mapped[str] = mapped_column(String, nullable=True)
+    __table_args__ = {'postgresql_partition_by': 'RANGE (created_at)'}  # noqa: RUF012
```

Comment:
`# noqa: RUF012` 

There's a confict between MyPy and Ruff. Ruff would like `__table_args__` and ` __mapper_args__`  to have the type `ClassVar`. But if `ClassVar` is added, then MyPy throws are error.